SampleFuzzingJobs/InPlaceFuzzing/inplacefuzzing/inplacefuzzing/inplacefuzzing.cpp

```diff
@@ -0,0 +1,110 @@
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+// File header definition
+struct Header {
+	char* constant;
+	char* searchString;
+	int numberOfReferenceFiles;
+};
+
+// File format definition
+struct FileFormat {
+	Header header;
+	char** files;
+};
+
+
+// Load file from the specified path
+// Do not check for any errors
+FileFormat loadFile(char* filePath) {
+	FILE* pSeed = fopen(filePath, "rb");
+
+	fseek(pSeed, 0, SEEK_END);
+	long size = ftell(pSeed);
+	rewind(pSeed);
+
+	char* buffer = (char*)malloc(sizeof(char) * size + 1);
+	memset(buffer, NULL, sizeof(char) * (size + 1));
+	fread(buffer, sizeof(char), size, pSeed);
+
+	Header header = { NULL, NULL, -1 };
+
+	char* pch = strtok(buffer, "\n");
+	while (pch != NULL)
+	{
+		int len = strlen(pch);
+
+		if (header.constant == NULL) {
+			header.constant = (char*)malloc(sizeof(char) * (len + 1));
+			memset(header.constant, NULL, len + 1);
+			memcpy(header.constant, pch, len);
+		}
+		else if (header.searchString == NULL) {
+			header.searchString = (char*)malloc(sizeof(char) * (len + 1));
+			memset(header.searchString, NULL, len + 1);
+			memcpy(header.searchString, pch, len);
+		}
+		else {
+			header.numberOfReferenceFiles = atoi(pch);
+			break;
+		}
+		pch = strtok(NULL, "\n");
+	}
+	FileFormat fileFormat = { header, (char**)malloc(sizeof(char*) * header.numberOfReferenceFiles)};
+	for (int i = 0; i < header.numberOfReferenceFiles; i++) {
+		pch = strtok(NULL, "\n");
+		int len = strlen(pch);
+		fileFormat.files[i] = (char*)malloc(sizeof(char) * (len + 1));
+		memset(fileFormat.files[i], NULL, len + 1);
+		memcpy(fileFormat.files[i], pch, len);
+	}
+	fclose(pSeed);
+	free(buffer);
+	return fileFormat;
+}
+
+int searchForStringInBuffer(char* buffer, char* searchString) {
+	int occurences = 0;
+	char* found = strstr(buffer, searchString);
+	int searchLen = strlen(searchString);
+	while (found != NULL) {
+		occurences++;
+		found = strstr(found + searchLen, searchString);
```

Comment:
Since the point of this sample is to show that we can find bugs I won't insist on using secure versions of the string functions ;-)